### PR TITLE
Update building in Linux docs

### LINKED ISSUE #333: [Doc] Please install CMake 3.14 or newer
Hi,

Just follow [Build using Docker](https://github.com/dotnet/runtime/blob/master/docs/workflow/building/coreclr/linux-instructions.md):

```
docker run --rm -v /home/zhaixiang/runtime:/runtime -w /runtime mcr.microsoft.com/dotnet-buildtools/prereqs:ubuntu-16.04-c103199-20180628134544 ./src/coreclr/build.sh
```

But CMake from mcr.microsoft.com/dotnet-buildtools/prereqs:ubuntu-16.04-c103199-20180628134544  is v3.51:

```
Commencing CoreCLR Repo build
__DistroRid: linux-x64
__RuntimeId: linux-x64
Setting up directories for build
Checking prerequisites...
3.5.1
Please install CMake 3.14 or newer from http://www.cmake.org/download/ or https://apt.kitware.com and ensure it is on your path.
```

Please update the document about docker.

Thanks,
Leslie Zhai

### PR DESCRIPTION
**Pull request by hoyosjs**
Fixes dotnet/runtime#333

### CODE REVIEW

**Comment by maryamariyan** [comment]
Also related to: https://github.com/dotnet/runtime/issues/304

@hoyosjs could you also please add a checkmark next to `docs/workflow/building/coreclr/linux-instructions.md` in https://github.com/dotnet/runtime/issues/304#issuecomment-558801558 if you cross-checked that this page is now up to date for dotnet/runtime?

**Comment by hoyosjs** [comment]
@maryamariyan I've further cleaned up. If approved I'll check the box.

**Comment by maryamariyan** [comment]
Thank you @hoyosjs

File: docs/workflow/building/coreclr/linux-instructions.md [hunk 1]
```diff
@@ -71,11 +73,9 @@ Add Kitware's APT feed to your configuration for a newer version of CMake. See t
 
 Install the following packages for the toolchain:
 
-- cmake
+- cmake (at least 3.14)
```
Comment (ViktorHofer) [inline docs/workflow/building/coreclr/linux-instructions.md]: the min version is 3.15.5

File: docs/workflow/building/coreclr/linux-instructions.md [hunk 2]
```diff
@@ -157,6 +153,6 @@ Running a single test
 After `src/coreclr/build-test.sh` is run, corerun from the Core_Root folder is ready to be run. This can be done by using the full absolute path to corerun, or by setting an environment variable to the Core_Root folder.
 
 ```sh
-export CORE_ROOT=~/runtime/artifacts/tests/coreclr/Linux.x64.Debug/Tests/Core_Root
+export CORE_ROOT=/runtime/artifacts/tests/coreclr/Linux.x64.Debug/Tests/Core_Root
```
Comment (ViktorHofer) [inline docs/workflow/building/coreclr/linux-instructions.md]: this is weird. I don't think any Unix user would place the runtime repo on the machine root.

**Review by jashook — APPROVED** [review]
Thanks for the change. Please make sure you address @ViktorHofer feedback.

File: docs/workflow/building/coreclr/linux-instructions.md (hunk 2 above)
Comment (hoyosjs) [reply docs/workflow/building/coreclr/linux-instructions.md]: Normally, no, but all our container builds and the docker instructions mount the repo there.

File: docs/workflow/building/coreclr/linux-instructions.md [hunk 3]
```diff
@@ -114,27 +114,23 @@ If you are using Fedora, then you will need to install the following packages:
...
 --------------------------------------
 
-To ensure that your system can allocate enough file-handles for the corefx build run `sysctl fs.file-max`. If it is less than 100000, add `fs.file-max = 100000` to `/etc/sysctl.conf`, and then run `sudo sysctl -p`.
-
-On Fedora:
-
-`$ sudo dnf install mono-devel`
+To ensure that your system can allocate enough file-handles for the libraries build run `sysctl fs.file-max`. If it is less than 100000, add `fs.file-max = 100000` to `/etc/sysctl.conf`, and then run `sudo sysctl -p`.
 
 Build the Runtime and Microsoft Core Library
```
Comment (danmosemsft) [inline docs/workflow/building/coreclr/linux-instructions.md]: I don't think "Microsoft Core Library" is a commonly used term. If you want you could change such string(s) to System.Private.Corelib.

File: docs/workflow/building/coreclr/linux-instructions.md (hunk 2 above)
Comment (hoyosjs) [reply docs/workflow/building/coreclr/linux-instructions.md]: @ViktorHofer are you ok with this? I don't feel strongly. I was just aligning to the instructions for the Docker mount.

Comment (ViktorHofer) [reply docs/workflow/building/coreclr/linux-instructions.md]: sure.

**Pull request merged**